homeassistant/components/geniushub/climate.py

```diff
@@ -0,0 +1,203 @@
+"""
+Supports Genius hub to provide climate controls.
+"""
+import logging
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    STATE_ECO, STATE_HEAT, STATE_AUTO, STATE_IDLE,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_OPERATION_MODE,
+    SUPPORT_ON_OFF, SUPPORT_AWAY_MODE)
+from . import GENIUS_HUB
+from homeassistant.const import (
```

Comment:
@MartinHjelmare I am a noob to Git. I have looked at: https://developers.home-assistant.io/docs/en/development_catching_up.html but when I run git fetch upstream dev I get the error: 'fatal: 'upstream' does not appear to be a git repository'.

When I run: script\gen_requirements_all.py I get the errors:
******* ERROR
Errors while importing:  homeassistant.components.climate.econet, homeassistant.components.google_assistant.auth
Make sure you import 3rd party libraries inside methods.

Please advise.

Reply:
Check if `upstream` is a known remote in your system.
```
git remote -v
```
This will show all remotes.
Add a remote like so.
```
git remote add [name] [url]
```
Eg
```
git remote add upstream https://github.com/home-assistant/home-assistant.git
```
Did you run `script/setup` first before running any other script?

Reply:
Usually we name things like this: `origin` is the name of your remote forked repo under your own github account. `upstream` is the name of the remote upstream "main" repo.

Reply:
I did run `script/setup `. I followed the instructions here:  https://developers.home-assistant.io/docs/en/development_environment.html. I am developing on Windows 10 under WSL. I followed the instructions for _Developing on Windows_, _Setup Local Repository_ and then _Setting up virtual environment_.

I know that something is not right with my setup as to commit the changes I need to copy the files out of WSL - which has to be wrong. Maybe this is the area I need to focus on and this is why I am running into these issues.

Thanks - will try this this evening. I also see that I need to address a hound issue.

Reply:
@GeoffAtHome I have resolved the 'hound issue' for you (and the merge conflicts).

@MartinHjelmare Any chance you can add me as a contributer (or what-what) to this PR, so I can help @GeoffAtHome?

Reply:
Only members can push directly to the branch of an open PR. You can open a PR to this branch in the fork though.

Reply:
I am a 'collaborator' on his repo - I pushed the latest changes, including a fix for these 'imported but unused' constants, above.

I can't mark this as resolved, and I wonder if it is an issue with permissions here, at your end?

Reply:
Resolved.

Reply:
Only organization members can push directly to the open PR branch.